kedro-datasets/pyproject.toml

```diff
@@ -237,7 +237,7 @@ test = [
     "openpyxl>=3.0.3, <4.0",
     "pandas-gbq>=0.12.0",
     "pandas>=2.0",
-    "Pillow~=9.0",
+    "Pillow~=10.0",
```

Comment:
This is less important/the fact that test requirements are manually specified like this is a separate issue that's not going to be solved in the scope of this PR.